extensions/spring-di/deployment/src/main/java/io/quarkus/spring/di/deployment/SpringDIProcessor.java

```diff
@@ -162,18 +162,24 @@ AnnotationsTransformerBuildItem beanTransformer(
     }
 
     /**
-     * Translate spring built in scope identifiers to CDI scopes.
+     * Translate spring built-in scope identifiers to CDI scopes.
      *
-     * @param clazz The class declaring the @Scope
+     * @param target The annotated element declaring the @Scope
      * @return A CDI built in (or session) scope that mostly matches
      *         the spring one. Websocket scope is currently mapped to @Dependant
      *         and spring custom scopes are not currently handled.
      */
-    private DotName getScope(final ClassInfo clazz) {
-        if (!clazz.annotations().containsKey(SPRING_SCOPE_ANNOTATION)) {
```

Comment:
Note: I've changed ClassInfo#annotations() (all levels annotations in the class) to ClassInfo#classAnnotation() (only class-level annotations of the class). The former was causing NPE when trying to access a method-level annotation value via clazz.classAnnotation().value()